File: 2452-words-within-two-edits-of-dictionary/2452-words-within-two-edits-of-dictionary.py
class Solution:
    def twoEditWords(self, queries: List[str], dictionary: List[str]) -> List[str]:
        res = []
        def checkDist(s: str, t: str) -> bool:
            dist = 0
            for i in range(len(s)):
                if s[i] != t[i]:
                    dist += 1
                if dist > 2:
                    return False
            return True
        for q in queries:
            for d in dictionary:
                if checkDist(q, d):
                    res.append(q)
                    break
        return res

File: 2452-words-within-two-edits-of-dictionary/2452-words-within-two-edits-of-dictionary.cpp
class Solution {
public:
    vector<string> twoEditWords(vector<string>& queries, vector<string>& dictionary) {
        vector<string> res;
        for(int i = 0; i < queries.size(); i++)
        {
            for(int j = 0; j < dictionary.size(); j++)
            {
                if(checkDist(queries[i], dictionary[j]))
                {
                    res.push_back(queries[i]);
                    break;
                }
            }
        }
        return res;
    }
public:
    bool checkDist(string& s, string& t)
    {
        char dist = 0;
        for(int i = 0; i < s.length(); i++)
        {
            if(s[i] != t[i])
            {
                dist++;
            }
            if(dist > 2)
            {
                return false;
            }
        }
        return true;
    }
};